std/mmfile.d

```diff
@@ -369,15 +372,24 @@ class MmFile
         }
         else version (Posix)
         {
-            // enforce shouldn't be here and close() is safe anyway
-            .close(fd);
+            version (linux)
+            {
+                if (file !is File.init)
+                {
+                    // The File destructor will close the file,
+                    // if it is the only remaining reference.
+                    return;
+                }
+            }
+            errnoEnforce(fd == -1 || fd <= 2
```

Comment:
Failure to close a fd which should be open indicates a dangling fd, which is a logic error... I think the response is apt, though a stack trace and better error message would be better, yeah.